pinot-controller/src/test/java/com/linkedin/pinot/controller/validation/StorageQuotaCheckerTest.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.controller.validation;
+
+import com.linkedin.pinot.common.config.AbstractTableConfig;
+import com.linkedin.pinot.common.config.QuotaConfig;
+import com.linkedin.pinot.common.config.SegmentsValidationAndRetentionConfig;
+import com.linkedin.pinot.controller.api.restlet.resources.TableSizeReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import junit.framework.Assert;
+import org.apache.commons.io.FileUtils;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+public class StorageQuotaCheckerTest {
+  TableSizeReader tableSizeReader;
+  AbstractTableConfig tableConfig;
+  private QuotaConfig quotaConfig;
+  private SegmentsValidationAndRetentionConfig validationConfig;
+  private final File TEST_DIR = new File(StorageQuotaCheckerTest.class.getName());
+
+  @BeforeClass
+  public void setUp() {
+    tableSizeReader = mock(TableSizeReader.class);
+    tableConfig = mock(AbstractTableConfig.class);
+    quotaConfig = mock(QuotaConfig.class);
+    validationConfig = mock(SegmentsValidationAndRetentionConfig.class);
+    when(tableConfig.getValidationConfig()).thenReturn(validationConfig);
+    when(validationConfig.getReplicationNumber()).thenReturn(2);
+    TEST_DIR.mkdirs();
```

Comment:
There is AfterClass method that is invoked even for errors. deleteOnExit() does not work if the directory is non-empty (which can happen for errors)